pkg/autodetect/main.go

```diff
@@ -81,12 +90,16 @@ func (b *Background) Stop() {
 func (b *Background) autoDetectCapabilities() {
 	apiList, err := b.availableAPIs()
 	if err != nil {
-		log.WithError(err).Info("Failed to determine the platform capabilities. Auto-detected properties will fallback to their default values.")
+		log.WithError(err).Info("failed to determine the platform capabilities, auto-detected properties will fallback to their default values.")
 		viper.Set("platform", v1.FlagPlatformKubernetes)
 		viper.Set("es-provision", v1.FlagProvisionElasticsearchNo)
 	} else {
-		// we could run all the detect* functions in parallel, but let's keep it simple for now
-		b.detectPlatform(apiList)
+
+		b.firstRun.Do(func() {
+			// the platform won't change during the execution of the operator, need to run it only once
+			b.detectPlatform(apiList)
```

Comment:
Does this affect the detection of Kafka and ES operators? 

Kafka or ES operators can be installed after Jaeger operator. We even removed OLM dependency on ES operator so we can expect people installing ES operator after Jaeger operator.

Reply:
No, the only part within the `firstRun.Do()` is the detect platform, which detects whether the operator is running in Kubernetes or OpenShift.

Kafka and ES are detected two lines below:

```golang
		b.detectElasticsearch(apiList)
		b.detectKafka(apiList)
```